src/platform/Zephyr/KeyValueStoreManagerImpl.h

```diff
@@ -0,0 +1,84 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Platform-specific key value storage implementation for Zephyr.
+ *
+ */
+
+#pragma once
+
+#include <zephyr.h>
+#include <fs/nvs.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace PersistedStorage {
+
+class KeyValueStoreManagerImpl final : public KeyValueStoreManager
+{
+    // Allow the KeyValueStoreManager interface class to delegate method calls to
+    // the implementation methods provided by this class.
+    friend class KeyValueStoreManager;
+
+public:
+    void Init();
+    
+    // NOTE: Currently this platform does not support partial and offset reads
```

Comment:
As we don't know if there will be any use-case that is a good idea to skip it.
Just wondering if we shouldn't assert as well, so that in future, we don't spend hours troubleshooting the problem if i.e. higher layers silently ignores error code or sth.

FYI: Reading with offset is implemented [here](https://github.com/nrfconnect/sdk-nrfxlib/blob/b35fc6728b5565b0d7046511df561f8151d7f106/nrf_security/src/psa/psa_eits_zephyr_settings.c#L176).

Reply:
Added an assert.